### DISCUSSION TITLE
iPXE cannot resolve external DNS only internal

### DISCUSSION THREAD

**Discussion by radamhu** [ipxe/ipxe#1074] [General]
netboot.xyz v2.0.73 192.168.0.11
pihole Pi-hole [v5.17.1] FTL [v5.23]  Web Interface [v5.20.1] 192.168.0.12
- DNS 192.168.0.12 (self)
- DHCP range 192.168.0.121-140

similar issue : https://github.com/ipxe/ipxe/discussions/846

I run netboot.xyz using the docker image, and I keep getting this error (https://ipxe.org/err/280a6011) which seems to be a DNS issue. 

![image](https://github.com/ipxe/ipxe/assets/13677208/9679f548-a416-45a8-8c1d-1633cc8b3326)

![image](https://github.com/ipxe/ipxe/assets/13677208/5913ed56-d98e-4130-9b70-a6aa039ee2b0)

if i check PIHOLE dns resolver, all queries give back the proper value 

![image](https://github.com/ipxe/ipxe/assets/13677208/221ae55d-394c-4321-a7ce-ff1d1ab432a1)

but at the iPXE client only internal DNS resolving works, external doesnt; so if a ping a local fqdn (pve01.home.local) its OK, but i.e. (external) index.hu isnt.

![Peek 2023-11-08 08-31](https://github.com/ipxe/ipxe/assets/13677208/ee4579db-562c-4b95-abb4-2c0f1992886e)

Any other intranet machine (with the same DHCP settings) can resolve both fqdn.

Any ideas?

**Comment by NiKiZe** [comment]
Use config command in your iPXE shell, check your gateway.
The last 2 lines of https://ipxe.org/280a6011 is really what you need to check.

**Reply by radamhu** [reply]
gw seems to have the right value

![image](https://github.com/ipxe/ipxe/assets/13677208/9a37edfe-a52d-4340-828c-47737db36665)

**Reply by NiKiZe** [reply]
TCP connection fails to open due to lack of working route.
You do have a IPv6 address and gateway as well, but not global routable address?, GitHub.com does currently not have any IPv6 address in its DNS records (but your pihole returns them).

Verify that your network has global working IPv6, if not it shouldn't have router advertisements enabled.
Alternatively, make sure your DNS don't give out non working AAAA records.

**Reply by radamhu** [reply]
neither my router or pihole serves dhcp/dns ipv6, they are disabled by default

BUT

if i hit "m" key to choose failsafe menu & configure network manually, external dns begin to works :)

![image](https://github.com/ipxe/ipxe/assets/13677208/72e3ead6-ee66-4aa2-b079-3ea9b69b8251)

![image](https://github.com/ipxe/ipxe/assets/13677208/0ebbfaea-9eb8-4752-a32e-cafb765dc578)

![image](https://github.com/ipxe/ipxe/assets/13677208/4bc681df-ef42-41cb-aa85-5b31adcda1cf)

![image](https://github.com/ipxe/ipxe/assets/13677208/597b2134-1ba1-4aa9-bf73-7e66d05b83a2)

ok, stg happens when iPXE configures its network automatically ...

**Reply by NiKiZe** [reply]
Take a look at your screenshots, you have gateway6, ip6, and dns6. Sure ip6 is auto-generated for link local, but you don't get gateway or dns6 unless something configures that.

In your pi log you also see the AAAA requests, but my bad because there it says NODATA as response. But maybe other responses comes in as well.

If it was me, I would start using tcpdump or wireshark to see which traffic is really flowing.

And just to clarify, happy to help with iPXE, but if you are using netbook.xyz specific features, please ask them directly. ;)
